### PR TITLE
641 feature request gitleak tool integration

### PR DESCRIPTION
**Pull request by chandelshubham63**
This **PR** is for new tool _**Gitleaks**_ integration in DDT, enhancing security testing capabilities. Gitleaks scans Git repositories for sensitive information leaks, providing crucial insights into potential security risks.
The integration allows users to specify the **repository path** and provides an **advanced mode** for additional **Git log** options and **report file path customization.**

**Testing-**
I tested it and it worked perfectly, ensuring accurate detection of sensitive information leaks.

**How to use-** 
To utilize, specify the repository path and optionally provide advanced settings for more detailed control.

### CODE REVIEW

**Review by bradleybogg — CHANGES_REQUESTED** [review]
Please resolve merge conflict before I review.

File: src/components/RouteWrapper.tsx [hunk 1]
```diff
@@ -386,6 +387,13 @@ export const ROUTES: RouteProperties[] = [
         description: "FTPconnect tool",
         category: "Miscellaneous",
     },
+    {
+        name: "Gitleaks",
+        path: "/tools/gitleaks",
+        element: <Gitleaks />,
+        description: "Detect hardcoded secrets in git repositories",
```
Comment (michaeljpigott) [inline src/components/RouteWrapper.tsx:394]: Need a full stop at the end of the sentence. 

- [ ] Add full stop.

**Review by michaeljpigott — CHANGES_REQUESTED** [review]
Hi there

This is looking really good. I only have one recommended change, which is to add a full stop to the description in RouteWrapper.tsx. 

Good work. Make this change and we'll arrange some application testing. 

Kind regards

Michael

**Comment by michaeljpigott** [comment]
Hey @chandelshubham63 

I am getting an error message even though the code has run. The message probably should be "Processed completed successfully". Can you please investigate? 
image

**Comment by chandelshubham63** [comment]
Hi @michaeljpigott, just wanted to clarify that `**gitleaks**` typically returns exit code 1 to indicate findings (leaks) in the repository, not necessarily an error. This is part of its normal behavior, as it's designed to alert users about potential sensitive information. So, while the process did complete, the exit code 1 signifies the presence of leaks that require attention rather than an error. Let me know if you have any further questions!

**Review by michaeljpigott — APPROVED** [review]
Hi @chandelshubham63 

Thank you for clarifying about the exit code. 

I have tested the tool in the app and appears to work well. 

Kind regards

Michael

**Pull request merged**